### PR TITLE
Allow customizing the set of default labels for stats exporting

### PR DESCRIPTION
**Pull request by ramonza**
* Introduce DefaultMonitoringLabels option
* Provide more documentation on how the exporter works with
  Stackdriver Monitoring concepts (labels, time series, resources).

Fixes: #16

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -56,3 +59,51 @@ func Example() {
 	// All outgoing requests from client will include a Stackdriver Trace header.
 	// See the ochttp package for how to handle incoming requests.
 }
+
+func Example_gKE() {
```
Comment (ramonza) [inline example_test.go]: This is the only spelling of GKE I could find to get godoc to render it correctly as GKE.

File: stats.go [hunk 2]
```diff
@@ -393,33 +402,36 @@ func namespacedViewName(v string) string {
...
-	labels[opencensusTaskKey] = taskValue
 	return labels
 }
 
-func newLabelDescriptors(keys []tag.Key) []*labelpb.LabelDescriptor {
-	labelDescriptors := make([]*labelpb.LabelDescriptor, len(keys)+1)
-	for i, key := range keys {
-		labelDescriptors[i] = &labelpb.LabelDescriptor{
+func newLabelDescriptors(defaults map[string]labelValue, keys []tag.Key) []*labelpb.LabelDescriptor {
+	labelDescriptors := make([]*labelpb.LabelDescriptor, 0, len(keys)+1)
```
Comment (songy23) [inline stats.go]: nit: s/len(keys)+1/len(keys)+len(defaults)

File: stats.go [hunk 3]
```diff
@@ -440,20 +452,27 @@ func equalMeasureAggTagKeys(md *metricpb.MetricDescriptor, m stats.Measure, agg
 		return fmt.Errorf("stackdriver metric descriptor was not created with aggregation type %T", agg.Type)
 	}
 
-	if len(md.Labels) != len(keys)+1 {
-		return errors.New("stackdriver metric descriptor was not created with the view labels")
-	}
-
 	labels := make(map[string]struct{}, len(keys)+1)
```
Comment (songy23) [inline stats.go]: ditto

**Comment by ramonza** [comment]
Thanks @songy23 updated, PTAL

File: stackdriver.go [hunk 4]
```diff
@@ -71,19 +98,52 @@ type Options struct {
...
+	//
+	// If you set DefaultMonitoringLabels, make sure that the Resource field
+	// together with these labels is unique to the
+	// current process. This is to ensure that there is only a single writer to
+	// each TimeSeries in Stackdriver.
+	//
+	// Set this to &Labels{} (a pointer to an empty Labels) to avoid getting the
+	// default "opencensus_task" label. You should only do this if you know that
+	// the Resource you set uniquely identifies this Go process.
+	DefaultMonitoringLabels *Labels
```
Comment (songy23) [inline stackdriver.go]: Suggestion: maybe also mention that unlike common labels, the label values of these default labels are fixed and that may potentially introduce high cardinality.

Comment (ramonza) [reply stackdriver.go]: What do you mean by common labels here? Why would being fixed introduce high cardinality - wouldn't it be the opposite (if they are not fixed)?

Comment (songy23) [reply stackdriver.go]: For example if users are using pid as the DefaultMonitoringLabels, then each process will only have one label value for it, while other labels (e.g method, caller) can be broken down by different label values.

File: example_test.go [hunk 5]
```diff
@@ -17,15 +17,18 @@ package stackdriver_test
...
 	"contrib.go.opencensus.io/exporter/stackdriver"
 	"contrib.go.opencensus.io/exporter/stackdriver/propagation"
 	"go.opencensus.io/plugin/ochttp"
 	"go.opencensus.io/stats/view"
 	"go.opencensus.io/trace"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
 )
 
-func Example() {
+func Example_defaults() {
```
Comment (rakyll) [inline example_test.go]: Example_NewExporter

**Pull request merged**